src/exception/call_stack/stackwalk.cr

```diff
@@ -0,0 +1,130 @@
+require "c/dbghelp"
+
+# :nodoc:
+struct Exception::CallStack
+  skip(__FILE__)
+
+  @@sym_loaded = false
+
+  def self.load_debug_info
+    return if ENV["CRYSTAL_LOAD_DEBUG_INFO"]? == "0"
+
+    unless @@sym_loaded
+      @@sym_loaded = true
+      begin
+        load_debug_info_impl
+      rescue ex
+        Crystal::System.print_exception "Unable to load debug information", ex
+      end
+    end
+  end
+
+  private def self.load_debug_info_impl
+    # TODO: figure out if and when to call SymCleanup (it cannot be done in
+    # `at_exit` because unhandled exceptions in `main_user_code` are printed
+    # after those handlers)
+    if LibC.SymInitializeW(LibC.GetCurrentProcess, nil, 1) == 0
+      raise RuntimeError.from_errno("SymInitializeW")
+    end
+    LibC.SymSetOptions(LibC.SymGetOptions | LibC::SYMOPT_LOAD_LINES | LibC::SYMOPT_FAIL_CRITICAL_ERRORS | LibC::SYMOPT_NO_PROMPTS)
+  end
+
+  def self.unwind
+    load_debug_info
+
+    # TODO: use stack if possible (must be 16-byte aligned)
+    context = Pointer(LibC::CONTEXT).malloc(1)
+    context.value.contextFlags = LibC::CONTEXT_FULL
+    LibC.RtlCaptureContext(context)
+
+    stack_frame = LibC::STACKFRAME64.new
+    stack_frame.addrPC.mode = LibC::ADDRESS_MODE::AddrModeFlat
+    stack_frame.addrFrame.mode = LibC::ADDRESS_MODE::AddrModeFlat
+    stack_frame.addrStack.mode = LibC::ADDRESS_MODE::AddrModeFlat
+
+    machine_type = LibC::IMAGE_FILE_MACHINE_AMD64
+    {% if flag?(:x86_64) %}
```

Comment:
Seems odd leaving this assignment outside of the actual branch that checks for the said arch.

```suggestion
    {% if flag?(:x86_64) %}
      machine_type = LibC::IMAGE_FILE_MACHINE_AMD64
```

Reply:
This is a workaround for #7975. Should probably be mentioned as a `TODO`.